components/site/src/sass/serde.rs

```diff
@@ -0,0 +1,481 @@
+    }
+}
+
+impl<'a> serde::ser::SerializeSeq for &'a mut SassMapSerializer {
+    // Must match the `Ok` type of the serializer.
+    type Ok = ();
+    // Must match the `Error` type of the serializer.
+    type Error = SassMapSerializerError;
+
+    // Serialize a single element of the sequence.
+    fn serialize_element<T>(&mut self, value: &T) -> SassMapSerializerResult
+    where
+        T: ?Sized + Serialize,
+    {
+        // output a comma for all but the first element
+        if !self.output.ends_with('[') {
+            self.output += ",";
+        }
+        value.serialize(&mut **self)
+    }
+
+    // Close the sequence.
+    fn end(self) -> SassMapSerializerResult {
+        self.output += "]";
+        Ok(())
+    }
+}
+
+// Same thing but for tuples.
+impl<'a> serde::ser::SerializeTuple for &'a mut SassMapSerializer {
+    type Ok = ();
+    type Error = SassMapSerializerError;
+
+    fn serialize_element<T>(&mut self, value: &T) -> SassMapSerializerResult
+    where
+        T: ?Sized + Serialize,
+    {
+        // output a comma for all but the first element
+        if !self.output.ends_with('[') {
+            self.output += ",";
+        }
+        value.serialize(&mut **self)
+    }
+
+    fn end(self) -> SassMapSerializerResult {
+        self.output += "]";
+        Ok(())
+    }
+}
+
+// Same thing but for tuple structs.
+impl<'a> serde::ser::SerializeTupleStruct for &'a mut SassMapSerializer {
+    type Ok = ();
+    type Error = SassMapSerializerError;
+
+    fn serialize_field<T>(&mut self, value: &T) -> SassMapSerializerResult
+    where
+        T: ?Sized + Serialize,
+    {
+        // output a comma for all but the first element
+        if !self.output.ends_with('[') {
+            self.output += ",";
+        }
+        value.serialize(&mut **self)
+    }
+
+    fn end(self) -> SassMapSerializerResult {
+        self.output += "]";
+        Ok(())
+    }
+}
+
+// Same thing but for tuple variants.
+impl<'a> serde::ser::SerializeTupleVariant for &'a mut SassMapSerializer {
+    type Ok = ();
+    type Error = SassMapSerializerError;
+
+    fn serialize_field<T>(&mut self, value: &T) -> SassMapSerializerResult
+    where
+        T: ?Sized + Serialize,
+    {
+        // output a comma for all but the first element
+        if !self.output.ends_with('[') {
+            self.output += ",";
+        }
+        value.serialize(&mut **self)
+    }
+
+    fn end(self) -> SassMapSerializerResult {
+        self.output += "]";
+        Ok(())
+    }
+}
+
+impl<'a> serde::ser::SerializeMap for &'a mut SassMapSerializer {
+    type Ok = ();
+    type Error = SassMapSerializerError;
+
+    /// The Serde data model allows map keys to be any serializable type. SCSS
+    /// only allows string keys so the implementation below will produce an invalid
+    /// SCSS map if the key serializes as something other than a string.  That
+    /// should be fine here as the source TOML only allows string keys as well.
```

Comment:
I will note that Sass maps do permit any value as a key: 
[Sass docs](https://sass-lang.com/documentation/values/maps/#:~:text=Maps%20allow%20any%20Sass%20values%20to%20be%20used%20as%20their%20keys)
[example in playground](https://connorskees.github.io/grassmeister/?v=YSB7CiAgICBjb2xvcjogaW5zcGVjdCgoMTogMiwgImEiOiAiYiIsIGM6IGQsIFsxLCAyLCAzXTogWzQsIDVdKSk7Cn0=)